db/commitlog/commitlog.cc

```diff
@@ -1963,6 +1963,17 @@ future<> db::commitlog::segment_manager::recalculate_footprint() {
         while (!_recycled_segments.empty()) {
             recycles.push_back(_recycled_segments.pop());
         }
+        // #9955 - must re-stock the queues before we do anything
+        // interruptable/continuation. Because both queues are
+        // used with push/pop eventually which _waits_ for signal
+        // but does _not_ verify that the condition is true once
+        // we return. So copy the objects and look at instead.
+        for (auto& filename : recycles) {
+            _recycled_segments.push(sstring(filename));
+        }
+        for (auto& s : reserves) {
+            _reserve_segments.push(sseg_ptr(s)); // you can have it back now.
+        }
```

Comment:
Looks like a preexisting problem, but still. queue::push() return false if the push failed, and here we aren't checking the return value. What happens if we fail to push?

Reply:
Ah, I see you just popped them off.

But I think you shouldn't be using seastar::queue here. This isn't decoupling some unrelated pieces of software, they are tightly coupled, and here we're losing by not seeing items waiting in push_eventually(). All we need is plain containers and maybe condition variables for wakeups.

Reply:
I agree, but it is a bigger change. This is a minimal fix. 
Also see seastar patch for queue, which I think could benefit from being somewhat more resilient to this (mis)usage pattern.